### PR TITLE
Adding CONTRIBUTING file.

### LINKED ISSUE #10: Creating a CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by GalOved**
Adding CONTRIBUTING file to the project.

Close #10

### CODE REVIEW

**Review by ohadrevach — APPROVED** [review]
Looks fine to me

**Review by GilShechter — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+After the installations typing 'vagrant up' and automatically the development environment will set up.
+
+
+### Issues
+
+---
+
+Issues should be used to report problems with the library, request a new feature, or to discuss potential changes before a PR is created.
+When you open issues use there [words](#https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests).Issues can also help the team know what every member of the team work on.
```
Comment (natifridman) [inline CONTRIBUTING.md:39]: These keywords should be used in the PR to link an issue, and not in the PR itself

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+3 Every PR should be linked to an issue.
+4 Each PR requires the approval of at least two team members.
+
+
+
+### Testing
+
+---
+
+Every PR is tested by Flake8.
```
Comment (natifridman) [inline CONTRIBUTING.md:74]: I would add that the CI tests passed and didn't fail

**Review by natifridman — COMMENTED** [review]
Nice work, added some comments.

I would also add something about the number of commits in the PR.

**Review by ohadrevach — APPROVED** [review]

**Comment by natifridman** [comment]
Please try not to force push during the review process. The reviewers can't see what you changed from the last time.
You can squash the commits after the PR is approved before the merge

**Review by efraimvis — APPROVED** [review]

**Review by WaseemAbuSobe — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+### Quicklinks
+
+---
+
+- [Pre-Requirements](#Pre-Requirements)
+- [Issues](#Issues)
+- [Pull Requests](#Pull Requests)
+- [Testing](#Testing)
+- [Code of Conduct](#Code of Conduct)
+- [Getting Help](#Getting Help)
```
Comment (liornoy) [inline CONTRIBUTING.md:19]: The hyperlinks seem broken here, from looking at the "rich diff"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+3 Create a new branch and new remote for the upstream repository.
+4 Make the changes on our local machine, add to the branch, commit the a massage and finally push the PR.
+5 Open a PR in our repository and follow this process: 
+	- Add a detailed description.
+	- Press on the Pull Requests.
+6 Link the issue to the PR when you open issues use there [words](#https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests).
+  Issues can also help the team know what every member of the team work on.
+
+
+###Pull Requests Standards
```
Comment (liornoy) [inline CONTRIBUTING.md:59]: missing space here

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+1 Create an [issue](#Issues).
+2 Fork a repository and clone the repository.
+3 Create a new branch and new remote for the upstream repository.
+4 Make the changes on our local machine, add to the branch, commit the a massage and finally push the PR.
+5 Open a PR in our repository and follow this process: 
+	- Add a detailed description.
+	- Press on the Pull Requests.
+6 Link the issue to the PR when you open issues use there [words](#https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests).
+  Issues can also help the team know what every member of the team work on.
```
Comment (liornoy) [inline CONTRIBUTING.md:56]: please fix newline spacing, looking at the rich diff

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+###Pull Requests Standards
+
+---
+Our standards for pull request is:
+
+1 Make polite notes to each other without using bad language.
+2 The PRs should be a meaningful and concise title without issue numbers or lines code.
+3 Every PR should be linked to an issue.
+4 Each PR requires the approval of at least two team members.
```
Comment (liornoy) [inline CONTRIBUTING.md:67]: please fix newline spacing, looking at the rich diff

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+Our standards To create a comfortable and positive environment we follow the following rules:
+
+- Using welcoming and inclusive language.
+- Showing empathy towards other community members.
+- Focusing on what is best for the community.
+- Being respectful of differing viewpoints and experiences.
+- Attempting collaboration before conflict.
+- Gracefully accepting constructive criticism.
+
+Our standards To avoid an unacceptable environment we follow the following rules:
```
Comment (liornoy) [inline CONTRIBUTING.md:103]: Those are rules of what NOT to follow.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
+##Contributing to Tasties project
```
Comment (liornoy) [inline CONTRIBUTING.md:1]: missing a space here

**Review by efraimvis — APPROVED** [review]

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+- Violence, threats of violence, or inciting others to commit self-harm.
+- The use of sexualized language or imagery and unwelcome sexual attention or advances.
+- Trolling, intentionally spreading misinformation, insulting/derogatory comments, and personal or political attacks.
+- Publishing others' private information, such as a physical or electronic address, without explicit permission.
+- Abuse of the reporting process to intentionally harass or exclude others.
+- Advocating for, or encouraging, any of the above behavior.
+
+#### Our Responsibilities
+
+Project maintainers are responsible for clarifying the standards of acceptable behavior and are expected to take appropriate and fair corrective
```
Comment (natifridman) [inline CONTRIBUTING.md:114]: Maybe worth to add somewhere in this doc who are the maintainers

**Comment by GalOved** [comment]
@natifridman I updated the CONTRIBUTING file according to the reviewes

File: CONTRIBUTING.md (hunk 6 above)
Comment (liornoy) [reply CONTRIBUTING.md:67]: @GalOved I think you missed this section :)
Please fix the spacing between the lines.

**Comment by liornoy** [comment]
@GalOved I think you missed the "Pull Requests Standards" section :)
Please fix the spacing between the lines.

**Comment by GalOved** [comment]
@liornoy l fix the "Pull Requests Standards" section.

**Comment by liornoy** [comment]
Please squash the commits so we'll remain with only one.
(As all of the current 4 commits are changing only 1 file)

**Comment by GalOved** [comment]
@liornoy i squash all the commits together.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+
+The maintainers are: 
+Lior Noy, Nati Fridman, Shay Vatarescu
+
+
+### Getting Help
+
+---
+
+If you need us you can contact on this e-mail
```
Comment (liornoy) [inline CONTRIBUTING.md:140]: Without an existing e-mail, this section doesn't make sense.

**Comment by GalOved** [comment]
@liornoy hi lior I fix the help section, delete that section becase it is not relvent for now.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+### Quicklinks
+
+---
+
+- [Pre-Requirements](#pre-requirements)
+- [Issues](#issues)
+- [Pull Requests](#pull-requests)
+- [Testing](#testing)
+- [Code of Conduct](#code-of-conduct)
+- [Getting Help](#getting-help)
```
Comment (liornoy) [inline CONTRIBUTING.md:19]: This quick link is not relevant anymore.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+
+
+### Pre Requirements
+
+---
+
+To contribute to our project please download and install the following:
+
+- Vagrant.
+- VirtualBox(or other VM that works with "Vagrant").
```
Comment (svatares) [inline CONTRIBUTING.md:29]: Maybe mention which version is most recommended to be used (hint, the one all of you are using)

Comment (svatares) [reply CONTRIBUTING.md:29]: Also, even better to add direct links to the downloads so it will be easier for a new contributor to start

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+
+After the installations type 'vagrant up' and automatically the development environment will set up.
```
Comment (svatares) [inline CONTRIBUTING.md:31]: This is vague, from which location does the vagrant up should run? (It is a basic thing that everyone should know, but if you weren't familiar with Vagrant, what would you do?)

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+
+
+### Issues
+
+---
+
+Issues should be used to report problems with the library, request a new feature, or to discuss potential changes before a PR is created.
+Please make sure that before the PR is created, the issue must be created and after the PR is created, the PR must be linked to the issue relevant to it.
```
Comment (svatares) [inline CONTRIBUTING.md:39]: This sentences are repeating themselves. Please rephrase it to make it more simplified a readable

Comment (svatares) [reply CONTRIBUTING.md:39]: Also, you're mentioning here the acronym `PR` where you only introduce it later on in the next section...

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+
+
+### Pull Requests
+
+---
+
+To create a pull request (PR) please follow the instructions:
+
+1. Create an [issue](#Issues).
```
Comment (svatares) [inline CONTRIBUTING.md:48]: Is this mandatory for any PR you're about to create?

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+2. Fork a repository and clone the repository.
```
Comment (svatares) [inline CONTRIBUTING.md:49]: Fork `the` repository, not `a`

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+3. Create a new branch and new remote for the upstream repository.
+4. Make the changes on our local machine, add to the branch, commit the a massage and finally push the PR.
```
Comment (svatares) [inline CONTRIBUTING.md:51]: ... `our` local machine -> ... `your` local machine 

And overall you're describing several actions on one bullet point, where it should probably only mention one action per bullet

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+	- Add a detailed description.
+	- Press on the Pull Requests.
+6. Link the issue to the PR when you open issues use there [words](#https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests).
+  Issues can also help the team know what every member of the team work on.
+
+
+### Pull Requests Standards
+
+---
+Our standards for pull request is:
```
Comment (svatares) [inline CONTRIBUTING.md:62]: ... pull request `is`: -> ... pull request `are`:

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+
+1. Make polite notes to each other without using bad language.
```
Comment (svatares) [inline CONTRIBUTING.md:64]: Can maybe rephrased to be more accurate and show a positive idea (instead of negative idea - `bad language`)

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+2. The PRs should be a meaningful and concise title without issue numbers or lines code.
```
Comment (svatares) [inline CONTRIBUTING.md:65]: ... `be a` meaningful ... -> ... `have a` ...

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our 
+community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, 
+gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.
+
+
+#### Our Standards
+
+
+Our standards To create a comfortable and positive environment we follow the following rules:
```
Comment (svatares) [inline CONTRIBUTING.md:94]: `Our standards` - does it really needed here?

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+
+- Using welcoming and inclusive language.
+- Showing empathy towards other community members.
+- Focusing on what is best for the community.
+- Being respectful of differing viewpoints and experiences.
+- Attempting collaboration before conflict.
+- Gracefully accepting constructive criticism.
+
+Our standards To avoid an unacceptable environment we ask not to do the following:
```
Comment (svatares) [inline CONTRIBUTING.md:103]: `Our standards` - does it really needed here?

File: CONTRIBUTING.md (hunk 9 above)
Comment (svatares) [reply CONTRIBUTING.md:114]: I can see that it's mentioned in the `Maintainers` section below

**Review by svatares — COMMENTED** [review]
Nice work, left a few comments

File: CONTRIBUTING.md (hunk 22 above)
Comment (GalOved) [reply CONTRIBUTING.md]: Hi, @svatares I corrected the change according to your suggestion and also rewrote part of the file.
You can see details about the things that were written in commit.

**Review by GilShechter — APPROVED** [review]
Looks great, nice job

**Comment by liornoy** [comment]
I think the CONTRIBUTING file looks good.

**Comment by liornoy** [comment]
@ohadrevach @WaseemAbuSobe Please go through the CONTRIBUTING file and review it,
and leave a comment if you agree/disagree with the content.
We must have everyone's consent before merging.

**Comment by svatares** [comment]
Nice work on changes, I resolved all of my comments, and most of the comments in general. @natifridman there are a couple of comments you left that were addressed, please see if those were resolved.
And in order to finish reviewing this PR we're missing @ohadrevach 's review

**Review by svatares — APPROVED** [review]

**Review by ohadrevach — APPROVED** [review]
looks good to me

**Review by natifridman — APPROVED** [review]
Approved
